linear_algebra/src/rank_of_matrix.py

```diff
@@ -0,0 +1,64 @@
+"""This is a python program that calculates the rank of a matrix"""
+""" BY - RUDRANSH BHARDWAJ"""
+
+
+def rank_of_matrix(matrix: list[list]) -> int:
+    """
+    Finds the rank of a matrix.
+    Args:
+        matrix (list of lists): The matrix as a list of lists.
+    Returns:
+        int: The rank of the matrix.
+    Example:
+    >>> matrix1 = [[1, 2, 3],
+    ...            [4, 5, 6],
+    ...            [7, 8, 9]]
+    >>> rank_of_matrix(matrix1)
+    2
+    >>> matrix2 = [[1, 0, 0],
+    ...            [0, 1, 0],
+    ...            [0, 0, 0]]
+    >>> rank_of_matrix(matrix2)
+    2
+    >>> matrix3 = [[1, 2, 3, 4],
+    ...            [5, 6, 7, 8],
+    ...            [9, 10, 11, 12]]
+    >>> rank_of_matrix(matrix3)
+    2
+    """
```

Comment:
add more unique doctests.

Reply:
ok

Reply:
removed my name and added more unique doctests